tests/functional/test_paramfile.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import logging
+
+from mock import patch, Mock, mock_open
+
+from awscli.testutils import FileCreator, BaseCLIWireResponseTest
+from awscli.clidriver import create_clidriver
+from botocore.vendored import requests
+
+logger = logging.getLogger(__name__)
+
+
+class FakeResponse(object):
+    def __init__(self, *args, **kwargs):
+        self.status_code = 200
+        self.text = 'function-name'
+
+
+class BaseTestCLIFollowParamURL(BaseCLIWireResponseTest):
```

Comment:
There are not too many tests that use `BaseCLIWireResponseTest`. I would recommend using `BaseAWSCommandParamsTest` instead as that class gives you a lot more to work with. For example you do not have to set a wire response, and instead you can set a parsed response.